apps/versions/urls.py

```diff
@@ -6,8 +6,10 @@
 
 urlpatterns = patterns(
     '',
-    url('^$', views.version_list, name='addons.versions'),
-    url('^format:rss$', VersionsRss(), name='addons.versions.rss'),
+    url('^(?:status:(?P<status>[^/]+))?$',
```

Comment:
You have extraneous `status:` in the `(?:status:(...))` regexps? It gives 404 for
- `/en-US/firefox/addon/foobar/versions/beta/`
- `/en-US/firefox/addon/foobar/versions/format:rss` (which might explain the test failure).

I'm not sure, but I think I would write it the longer (more explicit) way:
```
    url('^$',
        views.version_list, name='addons.versions'),
    url('^beta$',
        views.version_list, name='addons.versions', kwargs={'beta': True}),
    url('^format:rss$',
        VersionsRss(), name='addons.versions.rss'),
    url('^beta/format:rss$',
        VersionsRss(), name='addons.versions.rss', kwargs={'beta': True}),
```
And then change the `status` named parameter to `beta=False` in the `version_list` and `VersionRSS.get_object` views. This, however, prevents using just any status in the url, but I'm not sure that's needed?

If it is, you need to rewrite your urls to:
```
    url('^(?:(?P<status>[^/]+)/)format:rss$',
        VersionsRss(), name='addons.versions.rss'),
    url('^(?:(?P<status>[^/]+))?$',
        views.version_list, name='addons.versions'),
```
(Take care to have them in the correct order, or one will "shadow" the other, and `/versions/format:rss` won't work, and it believes `format:rss` is the status)